src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -131,14 +136,61 @@ create_distributed_table(PG_FUNCTION_ARGS)
 	if (distributionMethod != DISTRIBUTE_BY_HASH)
 	{
 		ConvertToDistributedTable(relationId, distributionColumnName,
-								  distributionMethod, INVALID_COLOCATION_ID);
+								  distributionMethod, INVALID_COLOCATION_ID,
+								  REPLICATION_MODEL_COORDINATOR);
 		PG_RETURN_VOID();
 	}
 
 	/* use configuration values for shard count and shard replication factor*/
 	CreateHashDistributedTable(relationId, distributionColumnName, ShardCount,
 							   ShardReplicationFactor);
 
+	if (ShouldSyncTableMetadata(relationId))
+	{
+		DistTableCacheEntry *cacheEntry = DistributedTableCacheEntry(relationId);
+
+		List *shardIntervalList = NIL;
+		List *commandList = NIL;
+		List *foreignConstraintCommands = NIL;
+		List *shardCreateCommandList = NIL;
+		ListCell *commandCell = NULL;
+		char *tableOwnerResetCommand = NULL;
+		char *metadataCommand = NULL;
+		char *truncateTriggerCreateCommand = NULL;
+
+		/* commands to create the table */
+		commandList = GetTableDDLEvents(relationId);
+
+		/* command to reset the table owner */
+		tableOwnerResetCommand = TableOwnerResetCommand(relationId);
+		commandList = lappend(commandList, tableOwnerResetCommand);
+
+		/* command to insert pg_dist_partition entry */
+		metadataCommand = DistributionCreateCommand(cacheEntry);
+		commandList = lappend(commandList, metadataCommand);
+
+		/* commands to create foreign key constraints */
+		foreignConstraintCommands = GetTableForeignConstraintCommands(relationId);
+		commandList = list_concat(commandList, foreignConstraintCommands);
+
+		/* commands to create the truncate trigger of the mx table */
+		truncateTriggerCreateCommand = TruncateTriggerCreateCommand(relationId);
+		commandList = lappend(commandList, truncateTriggerCreateCommand);
+
+		/* commands to insert pg_dist_shard & pg_dist_shard_placement entries */
+		shardIntervalList = LoadShardIntervalList(relationId);
+		shardCreateCommandList = ShardListInsertCommand(shardIntervalList);
+		commandList = list_concat(commandList, shardCreateCommandList);
```

Comment:
It might be nice to capture all these commands in a `GetDistributedTableDDLEvents` function like we have in MX.

unrelated note: shardCreate makes me think of shard creation, should probably be shardInsert.